semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/callgraph/DynamicCallLinker.scala

```diff
@@ -46,11 +46,12 @@ class DynamicCallLinker(cpg: Cpg) extends CpgPass(cpg) {
 
   /** Main method of enhancement - to be implemented by child class
     */
-  override def run(): Iterator[DiffGraph] = {
-    val dstGraph = DiffGraph.newBuilder
+  override def run(dstGraph: DiffGraphBuilder): Unit = {
     // Perform early stopping in the case of no virtual calls
-    if (!cpg.call.exists(_.dispatchType == DispatchTypes.DYNAMIC_DISPATCH)) return Iterator()
-    else initMaps()
+    if (!cpg.call.exists(_.dispatchType == DispatchTypes.DYNAMIC_DISPATCH)){
+      return;
+    }
+    initMaps()
     // ValidM maps class C and method name N to the set of
     // func ptrs implementing N for C and its subclasses
     cpg.typeDecl
```

Comment:
I know this isn't part of your PR, but since I just had this file in front of me, IMO this is more readable:
```
    for {
      c <- cpg.typeDecl
      n <- c._methodViaAstOut
    } {
      val methodName = n.fullName
      val candidates = allSubclasses(c.fullName).flatMap(staticLookup(_, n))
      validM.put(methodName, candidates)
    }
```